lucet-runtime/lucet-runtime-tests/src/timeout.rs

```diff
@@ -273,6 +275,21 @@ macro_rules! timeout_tests {
             t.join().unwrap();
         }
 
+        // after resetting, existing kill swtiches will not work. we can still run a function.
+        // FIXME KTM 2020-02-19: I did not see a way around this behavior. Bring this up in review.
```

Comment:
I think it would end up surprising if a `KillSwitch` from some prior `reset` were to still work, especially since they have a `operates only once` notion when they _do_ work. Being able to reuse an old `KillSwitch` to reportedly `NotTerminable` an instance is also not what I imagine would be expected either, so maybe it makes sense to have a `KillError::Invalid` to communicate having used a stale resource?

Reply:
Yeah, this totally makes sense! Because `Invalid` might be vague to a caller, I also added some doc comments to `KillError` to explain what we mean here.